src/main/java/duke/taskTypes/Deadline.java

```diff
@@ -0,0 +1,62 @@
+package duke.taskTypes;
+
+import duke.exception.DukeException;
+
+import duke.exception.EmptyTimeException;
+import duke.exception.InvalidFormatException;
+import duke.exception.EmptyDescriptionException;
+
+import java.util.List;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/**
+ * Deadline Task class that sets description of task, date, time
+ */
+public class Deadline extends Task{
+
+    /**
+     * Takes in a string and splits msg into based on /by pattern. Set the eventType and time of the instance
+     *
+     * @param input string from the user
+     */
+    public Deadline(String input, boolean isDone) throws DukeException {
+        super(isDone);
+        List<String> results = Pattern.compile("/by").splitAsStream(input).map(x->x.trim()).collect(Collectors.toList());
```

Comment:
I don't know if its GitHub or your code but the indentation for this seems a bit off, maybe you can consider splitting it at each new function so that its easier to read?
E.g:
~~~
    Pattern.compile("/by")
            .splitAsStream(input)
~~~